src/generate/CodeModelAzImpl.ts

```diff
@@ -817,7 +818,16 @@ export class CodeModelCliImpl implements CodeModelAz {
     }
 
     public get Extension_Mode(): string {
-        return AzConfiguration.getValue(CodeGenConstants.extensionMode);
+        let extensionMode = AzConfiguration.getValue(CodeGenConstants.extensionMode);
+        this.codeModel.operationGroups.forEach((operationGroup) => {
+            if (
+                operationGroup.language['az'].command === this.Extension_Name &&
+                !isNullOrUndefined(operationGroup.language?.['cli']?.extensionMode)
+            ) {
+                extensionMode = operationGroup.language?.['cli']?.extensionMode;
```

Comment:
Is it needed to set extensionMode multiple times for each operationGroup?
Can we break after extensionMode is set?

Reply:
Actually we can't, this is an override mechanism, If the operationGroup has been renamed the same as extension name, then the extension mode of that extension should also be override by that operationGroup's mode